frame/currency-factory/src/ranges.rs

```diff
@@ -0,0 +1,191 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+pub use composable_traits::currency::RangeId;
+use frame_support::{traits::Get, BoundedVec};
+use scale_info::TypeInfo;
+use sp_runtime::{
+	traits::{CheckedAdd, Saturating},
+	ArithmeticError, DispatchError,
+};
+pub struct MaxRanges;
+
+impl Get<u32> for MaxRanges {
```

Comment:
very dynamic and extensible.

Reply:
ok. so i always think that as of risk.

Reply:
Not sure if I fully understand the comment. It can definitely be passed down from the config trait of the pallet, although I do not see a usecase for more that ~5 ranges to be honest. In this case we just put a reasonable upper bound on it. If we wanted to upgrade that bound, we'd need a runtime upgrade anyways.

Reply:
I'd also prefer not to pollute the signatures too much. With the introduction of MaxEncodedLen, many storage items previously using Vec will need to decide on an upper bound.